src/node_http2.cc

```diff
@@ -2184,6 +2184,17 @@ ssize_t Http2Stream::Provider::Stream::OnRead(nghttp2_session* handle,
 
   size_t amount = 0;          // amount of data being sent in this data frame.
 
+  // Remove all empty chunks from the head of the queue.
+  // This is done here so that .write('', cb) is still a meaningful way to
+  // find out when the HTTP2 stream wants to consume data, and because the
+  // StreamBase API allows empty input chunks.
+  while (!stream->queue_.empty() && stream->queue_.front().buf.len == 0) {
```

Comment:
One question, if we remove the empty buffer and do nothing, will the header be sent?

Refs: https://github.com/nodejs/node/pull/18673/files#diff-696b2cc418addca5f3fe5020058f8b15R1622

Reply:
@XadillaX Hm – I’m not sure I can wrap my head around everything, but this code here is only ever called by nghttp2 after the headers were sent anyway… does that answer your question?

So, yes, writing an empty buffer would trigger the headers to be sent, as I understand it.